File: src/baseDatos/Serializador.java
package baseDatos;

import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectOutputStream;
import java.util.ArrayList;

import gestorAplicacion.gestorMusica.Cancion;
import gestorAplicacion.gestorMusica.Coleccion;
import gestorAplicacion.gestorMusica.Lista;
import gestorAplicacion.gestorMusica.MeGusta;
import gestorAplicacion.gestorPersonas.Artista;
import gestorAplicacion.gestorPersonas.Usuario;

public class Serializador {
	
	public static <E> void serializar(ArrayList<E> lista, String nombre) {
		
		FileOutputStream fileOut;

		try {
			String path = System.getProperty("user.dir") + "/src/baseDatos/temp/" + nombre + ".txt";
			fileOut = new FileOutputStream(path);
			ObjectOutputStream out = new ObjectOutputStream(fileOut);
			out.writeObject(lista);
			// System.out.println("Se guardó correctamente en " + nombre);
			out.close();
			fileOut.close();
			
		} catch (FileNotFoundException e) {
			e.printStackTrace();
		} catch (IOException e) {
			e.printStackTrace();
		}
	}
	
	public static void serializarDatos() {
		Serializador.serializar(Usuario.getUsuariosExistentes(), "Usuarios");
		Serializador.serializar(Artista.getArtistasDisponibles(), "Artistas");
		Serializador.serializar(Cancion.getCancionesDisponibles(), "Canciones");
		Serializador.serializar(Coleccion.getColeccionesExistentes(), "Colecciones");
		Serializador.serializar(Lista.getListasExistentes(), "Listas");
		Serializador.serializar(MeGusta.getFavoritosExistentes(), "MeGusta");
		
	}
}

File: Python/src/baseDatos/serializador.py
''' Módulo serializador.py
    Autores: Carolina Álvarez Murillo, Miller Johan Chica Acero,
             Tomás Rodríguez Taborda, Jerónimo Ledesma Patiño,
             Catalina Restrepo Salgado
    Este módulo contiene la clase Serializador, que nos va a permitir guardar en cada archivo .txt un 
    flujo de bytes correspondiente a los objetos que se han creado durante la ejecución del programa
'''

import pickle
import pathlib
import os

from gestorAplicacion.gestorMusica.cancion import Cancion
from gestorAplicacion.gestorMusica.coleccion import Coleccion
from gestorAplicacion.gestorMusica.lista import Lista
from gestorAplicacion.gestorMusica.meGusta import meGusta
from gestorAplicacion.gestorPersonas.artista import Artista
from gestorAplicacion.gestorPersonas.usuario import Usuario

class Serializador:

    def serializar(lista, nombre):
        ruta = os.path.join(pathlib.Path(__file__).parent.absolute(), "temp/" + nombre + ".txt")
        
        try:
            picklefile = open(ruta, 'wb')
            pickle.dump(lista, picklefile)
            picklefile.close()
            
        except:
            print("¡Ocurrió un error!")

    def serializarDatos():
        Serializador.serializar(Usuario.getUsuariosExistentes(), "Usuarios")
        Serializador.serializar(Artista.getArtistasDisponibles(), "Artistas")
        Serializador.serializar(Cancion.getCancionesDisponibles(), "Canciones")
        Serializador.serializar(Coleccion.getColeccionesExistentes(), "Colecciones")
        Serializador.serializar(Lista.getListasExistentes(), "Listas")
        Serializador.serializar(meGusta.getFavoritosExistentes(), "MeGusta")
        